app/controllers/application_controller.rb

```diff
@@ -10,4 +10,10 @@ def authentication_callback
   def authenticate!
     redirect_to '/auth/nyulibraries' unless session[:auth] && (session[:auth]['credentials']['expires'] == false || session[:auth]['credentials']['expires_at'] > Time.now.to_i)
```

Comment:
Hi @pachonk ,
You might want to create a private helper method to encapsulate the logic after `unless`.
i.e., 
```
def valid_session?
  session[:auth] && 
    (session[:auth]['credentials']['expires'] == false || 
      session[:auth]['credentials']['expires_at'] > Time.now.to_i)
end
```
then your `authenticate!` method could be:
```
...
  redirect_to '/auth/nyulibraries' unless valid_session?
...
```